nautobot/docs/development/core/settings.md

```diff
@@ -0,0 +1,82 @@
+* Care needs to be taken when implementing environment variable support for settings that are also Constance-enabled - you must check for `"<variable_name>" in os.environ` before setting a variable in the Nautobot settings, so that it remains unset (and therefore manageable by Constance) if the environment variable isn't specified, rather than the simpler common pattern of just `SETTING = os.getenv("<variable_name>", <default value>)` used elsewhere in the settings.
+
+## Environment Variable Support
+
+For all settings where it's feasible to do so, you should make sure that it's configurable via environment variable as an alternative to directly editing `nautobot_config.py`. Exceptions would be for settings that cannot easily be defined as an environment variable string, such as Python functions, complex dicts, etc.
+
+The environment variable should typically be named `NAUTOBOT_<setting_name>` unless there is a *strong* existing convention in other tools for a different name.
+
+For Constance-enabled settings, the general pattern would be:
+
+```python
+if "NAUTOBOT_MY_SETTING" in os.environ and os.environ["NAUTOBOT_MY_SETTING"] != "":
+    MY_SETTING = os.environ["NAUTOBOT_MY_SETTING"]
+```
+
+and in `CONSTANCE_CONFIG`:
+
+```python
+CONSTANCE_CONFIG = {
+    "MY_SETTING": ConstanceConfigItem(
+        default="<default_value>",
+        help_text="<user-facing string>",
+    ),
+}
+```
+
+For non-Constance-enabled settings, the pattern is simpler:
+
+```python
+MY_SETTING = os.getenv("NAUTOBOT_MY_SETTING", "<default_value>")
+```
+
+## Documentation
+
+As of Nautobot 2.2.0, settings documentation is automatically generated from the contents of `nautobot/core/settings.yaml`. This document describes a JSON Schema for Nautobot settings, with a number of custom extensions to support richer documentation. In general, documentation for a typical string-based setting will take the following form:
+
+```yaml
+"properties":
+  "MY_SETTING":
+    "default": "some_default"
+    "description": "My setting. One to three sentences here at most."
+    "details": "<optional additional details, examples, caveats, etc.>"
+    "environment_variable": "NAUTOBOT_MY_SETTING"
+    "see_also":
+      "Documentation on example.com": "http://docs.example.com/my-setting"
+    "type": "string"
+    "version_added": "2.2.0"
+```
+
+As with a standard JSON Schema, it's possible to define other data types (`"type": "array"`, `"type": "object"`, etc.) and those will have other available/expected keys. Refer to documents like [Creating your first schema](https://json-schema.org/learn/getting-started-step-by-step) for details if you're less familiar with JSON Schema conventions.
+
+The special keys added specifically for documentation are as follows:
+
+* `details`: additional text beyond what's included in `description`, including examples, caveats, etc.
```

Comment:
Mention that the details and default_literal fields support markdown in our default implementation?